prediction_market_agent/agents/microchain_agent/blockchain/contract_class_converter.py

```diff
@@ -0,0 +1,143 @@
+import json
+from typing import Any
+
+import requests_cache
+from eth_typing import ChecksumAddress
+from microchain import Function
+from prediction_market_agent_tooling.tools.contract import ContractOnGnosisChain
+from web3 import Web3
+
+from prediction_market_agent.agents.microchain_agent.blockchain.code_interpreter import (
+    CodeInterpreter,
+)
+from prediction_market_agent.agents.microchain_agent.blockchain.models import (
+    AbiItemTypeEnum,
+    ABIMetadata,
+)
+
+TYPE_MAPPING: dict[str, (str, Any)] = {
+    "address": (
+        "str",
+        Web3.to_checksum_address("0xe91d153e0b41518a2ce8dd3d7944fa863463a97d"),
+    ),  # Solidity address to Python string
+    "uint": ("int", 1),  # Solidity unsigned integer to Python int
+    "uint8": ("int", 1),  # Solidity unsigned integer to Python int
+    "uint256": ("int", 1),  # Solidity unsigned integer (256-bit) to Python int
+    "int": ("int", 1),  # Solidity integer to Python int
+    "bool": ("bool", True),  # Solidity boolean to Python bool
+    "string": ("str", "test string"),
+    # Add more mappings as needed
+}
+
+
+class ClassFactory:
+    def create_class(self, class_name, base_classes=(), attributes=None):
+        if attributes is None:
+            attributes = {}
+
+        new_class = type(class_name, base_classes, attributes)
+        return new_class
+
+
+class ContractClassConverter:
+    """Class responsible for reading a smart contract on Gnosis Chain and converting its functionalities into Python classes."""
+
+    def __init__(self, contract_address: ChecksumAddress):
+        # For caching requests of the same contract
+        self.session = requests_cache.CachedSession(backend="sqlite")
+        self.contract_address = contract_address
+
+    def fetch_from_blockscout(self) -> dict[str, Any]:
+        r = self.session.get(
+            f"https://gnosis.blockscout.com/api/v2/smart-contracts/{self.contract_address}"
+        )
+        r.raise_for_status()
+        data: dict[str, Any] = r.json()
+        return data
+
+    def get_abi(self) -> list[ABIMetadata]:
+        data = self.fetch_from_blockscout()
+        contract_abi = data["abi"]
+        # We extract only functions, not events
```

Comment:
Can you explain why?

Reply:
Goal of this PR is to generate wrapper-functions for the functions available in the smart contract. Since events are not callable, I neglect them.